evap/staff/forms.py

```diff
@@ -650,13 +650,18 @@ def save(self, commit=True):
 class QuestionForm(forms.ModelForm):
     class Meta:
         model = Question
-        fields = ('order', 'questionnaire', 'text_de', 'text_en', 'type')
+        fields = ('order', 'questionnaire', 'text_de', 'text_en', 'type', 'allow_textanswers')
         widgets = {
             'text_de': forms.Textarea(attrs={'rows': 2}),
             'text_en': forms.Textarea(attrs={'rows': 2}),
             'order': forms.HiddenInput(),
         }
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.instance.pk and self.instance.type in [Question.TEXT, Question.HEADING]:
+            self.fields['allow_textanswers'].disabled = True
```

Comment:
No custom clean method: If users change the type _and_ the allow_textanswers on the form, they can submit successfully (however, the save method will set allow_textanswers back to False). I think they should get an error. (JS should disable the checkbox for them, but django will not verify this.)

Reply:
my thoughts were that this part here disables the checkboxes on page load if the initial question type is `TEXT` or `HEADING`, so for questions that don't allow additional text answers.
the same happens via js when these types are selected when editing the form. in combination, this disallows selecting one of these types together with additional text answers in the ui. so there should be no case where any errors are shown. i think forcing correct input data in the ui is better than showing an error afterward.

we could then clean the data after submit to prevent any form data fiddling or - as i implemented it - always set `allow_textanswers` to `False` on saving the model if one of the disallowed types was chosen. you don't seem to like the current implementation, what would you prefer instead?
it is a staff form, so we don't have to fear people manually editing post data. by extending the save method, a consistent state is enforced on django side. we could also implement an additional database constraint, but that would still require the other things if we don't want to run into a database error on saving.
i think a falsely set `True` value for other question types in the end also would not break anything because `allow_textanswers` is only evaluated for rating questions.

Reply:
Hmm, I think enforcing on the frontend and in safe() is okay -- however, I still think that having only the check in `save` is a bit too weak and would vote in favor of also adding a [database constraint](https://docs.djangoproject.com/en/3.0/ref/models/constraints/#module-django.db.models.constraints).